Code/Legacy/CrySystem/DebugCallStack.cpp

```diff
@@ -153,13 +154,13 @@ void DebugCallStack::SetUserDialogEnable(const bool bUserDialogEnable)
 DWORD g_idDebugThreads[10];
 const char* g_nameDebugThreads[10];
 int g_nDebugThreads = 0;
-volatile int g_lockThreadDumpList = 0;
+AZStd::spin_mutex g_lockThreadDumpList = 0;
```

Comment:
Well this is a bug:
The mutex is being assigned to 0.
```suggestion
AZStd::spin_mutex g_lockThreadDumpList;
```